### PR TITLE
Added eth withdrawal ack notifications

### PR DESCRIPTION
**Pull request by Mingela**
- Added withdrawal ack RMQ publishing (notification service will react)
- Yes, it takes a lot of time to react, so sleep is so long (at least locally)
- Time slept will be reduced with removing obsolete modules and its tests

### CODE REVIEW

File: eth/src/main/kotlin/com/d3/eth/abi/AbiGsonHelper.kt [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+    }
+
+    private class ByteArrayToEthHexTypeAdapter : JsonSerializer<ByteArray>, JsonDeserializer<ByteArray> {
+        @Throws(JsonParseException::class)
+        override fun deserialize(
+            json: JsonElement,
+            typeOfT: Type,
+            context: JsonDeserializationContext
+        ): ByteArray {
+            return DatatypeConverter.parseHexBinary(removePrefixIfNeeded(json.asString))
```
Comment (rkharisov) [inline eth/src/main/kotlin/com/d3/eth/abi/AbiGsonHelper.kt]:
```suggestion
            return org.web3j.utils.Numeric.hexStringToByteArray(json.asString)
```
Use this method instead, this allows you to remove custom methods for removing / adding a prefix

File: eth/src/main/kotlin/com/d3/eth/abi/AbiGsonHelper.kt [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+        }
+
+        override fun serialize(
+            src: ByteArray,
+            typeOfSrc: Type,
+            context: JsonSerializationContext
+        ): JsonElement {
+            return JsonPrimitive(addPrefixIfNeeded(DatatypeConverter.printHexBinary(src)))
```
Comment (rkharisov) [inline eth/src/main/kotlin/com/d3/eth/abi/AbiGsonHelper.kt]:
```suggestion
            return JsonPrimitive(org.web3j.utils.Numeric.toHexString(src))
```

File: eth/src/main/kotlin/com/d3/eth/sidechain/EthChainHandler.kt [hunk 3]
```diff
@@ -162,6 +170,54 @@ class EthChainHandler(
...
+     */
+    private fun handleWithdrawal(
+        transaction: Transaction,
+        time: BigInteger
+    ): List<SideChainEvent.PrimaryBlockChainEvent> {
+        logger.info { "Handle Ethereum master contract call tx ${transaction.hash}" }
+
+        val receipt = web3.ethGetTransactionReceipt(transaction.hash).send()
+
+        if (!receipt.transactionReceipt.get().isStatusOK) {
```
Comment (rkharisov) [inline eth/src/main/kotlin/com/d3/eth/sidechain/EthChainHandler.kt:186]: Can we be sure transaction receipt is not empty here?

File: eth/src/test/kotlin/com/d3/eth/abi/AbiDecoderTest.kt [hunk 4]
```diff
@@ -0,0 +1,131 @@
...
..."addToken","outputs":[],"payable":false,"stateMutability":"nonpayable","type":"function"},{"constant":true,"inputs":[],"name":"owner_","outputs":[{"name":"","type":"address"}],"payable":false,"stateMutability":"view","type":"function"},{"constant":false,"inputs":[{"name":"tokenAddress","type":"address"},{"name":"amount","type":"uint256"},{"name":"to","type":"address"},{"name":"txHash","type":"bytes32"},{"name":"v","type":"uint8[]"},{"name":"r","type":"bytes32[]"},{"name":"s","type":"bytes32[]"},{"name":"from","type":"address"}],"name":"withdraw","outputs":[],"payable":false,"stateMutability":"nonpayable","type":"function"},{"inputs":[{"name":"initialPeers","type":"address[]"}],"payable":false,"stateMutability":"nonpayable","type":"constructor"},{"payable":true,"stateMutability":"payable","type":"fallback"},{"anonymous":false,"inputs":[{"indexed":false,"name":"ethereumAddress","type":"address"},{"indexed":false,"name":"accountId","type":"bytes"}],"name":"IrohaAccountRegistration","type":"event"},{"anonymous":false,"inputs":[{"indexed":false,"name":"asset","type":"address"},{"indexed":false,"name":"recipient","type":"address"}],"name":"InsufficientFundsForWithdrawal","type":"event"}]
```
Comment (rkharisov) [inline eth/src/test/kotlin/com/d3/eth/abi/AbiDecoderTest.kt]: Let's move file in test resources. No need to hold such big string in test code

File: notary-eth-integration-test/src/integration-test/kotlin/integration/eth/WalletWithdrawalPipelineIntegrationTest.kt [hunk 5]
```diff
@@ -229,6 +240,15 @@ class WalletWithdrawalPipelineIntegrationTest {
                 masterBalanceInitial.subtract(bigIntegerValue),
                 masterBalanceActual
             )
+
+            Thread.sleep(15_000)
```
Comment (rkharisov) [inline notary-eth-integration-test/src/integration-test/kotlin/integration/eth/WalletWithdrawalPipelineIntegrationTest.kt:244]: I hope the removal of obsolete modules will not take too much time  😊

**Review by rkharisov — CHANGES_REQUESTED** [review]

File: notary-eth-integration-test/src/integration-test/kotlin/integration/eth/WalletWithdrawalPipelineIntegrationTest.kt (hunk 5 above)
Comment (Mingela) [reply notary-eth-integration-test/src/integration-test/kotlin/integration/eth/WalletWithdrawalPipelineIntegrationTest.kt:244]: Yes it should be done soon :)

File: eth/src/test/kotlin/com/d3/eth/abi/AbiDecoderTest.kt (hunk 4 above)
Comment (Mingela) [reply eth/src/test/kotlin/com/d3/eth/abi/AbiDecoderTest.kt]: Okay

File: eth/src/main/kotlin/com/d3/eth/sidechain/EthChainHandler.kt (hunk 3 above)
Comment (Mingela) [reply eth/src/main/kotlin/com/d3/eth/sidechain/EthChainHandler.kt:186]: Yes, since we've got this transaction from block

**Review by rkharisov — APPROVED** [review]

**Review by Alexey-N-Chernyshov — APPROVED** [review]

**Pull request merged**